clueless/state.py

```diff
@@ -41,33 +51,63 @@ def sock_addr(self) -> str:
 
     @property
     def is_registered(self) -> bool:
-        return self.character is not None
+        return self.state is UserState.REGISTERED
```

Comment:
I'd keep this as `self.character is not None`. That state isn't really necessary, the `character` starts out as `None` and is only assigned a `Character` value when the user registers. Could also be done with `self.display_name`, which is `None` until the player registers.